packages/core/tests/integration/components/navi-visualization-config/goal-gauge-test.ts

```diff
@@ -0,0 +1,98 @@
+import { setupRenderingTest } from 'ember-qunit';
+import { render, fillIn, blur, findAll } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { TestContext as Context } from 'ember-test-helpers';
+import StoreService from '@ember-data/store';
+import NaviVisualizationConfigGoalGaugeComponent from 'navi-core/components/navi-visualization-config/goal-gauge';
+
+const Template = hbs`
+  <NaviVisualizationConfig::GoalGauge
+    @response={{this.response}}
+    @request={{this.request}}
+    @options={{this.options}}
+    @onUpdateConfig={{this.onUpdateConfig}}
+  />`;
+
+type ComponentArgs = NaviVisualizationConfigGoalGaugeComponent['args'];
+
+interface TestContext extends Context, ComponentArgs {}
+
+module('Integration | Component | visualization config/goal gauge', function(hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function(this: TestContext) {
+    const store = this.owner.lookup('service:store') as StoreService;
+    this.options = {
+      baselineValue: 200,
+      goalValue: 0,
+      metricCid: 'cid_bubbles'
+    };
+    this.onUpdateConfig = () => null;
+    this.request = store.createFragment('bard-request-v2/request', {
+      columns: [
+        {
+          cid: 'cid_bubbles',
+          type: 'metric',
+          field: 'bubbles',
+          parameters: {},
+          alias: "Glass Bottles of the ranch's finest pasteurized whole milk!!!!!!!",
+          source: 'bardOne'
+        }
+      ],
+      filters: [],
+      sorts: [],
+      requestVersion: '2.0',
+      dataSource: 'bardOne',
+      table: 'network'
+    });
+    this.response = { rows: [{ bubbles: 32 }], meta: {} };
+  });
+
+  test('it renders', async function(this: TestContext, assert) {
+    await render(hbs`<NaviVisualizationConfig::GoalGauge/>`);
+
+    const headers = findAll('.goal-gauge-config__section-header').map(el => el?.textContent?.trim());
+    assert.deepEqual(headers, ['Label', 'Baseline', 'Goal'], 'headers are displayed for goal gauge config');
+  });
+
+  test('onUpdateConfig baselineValue input', async function(this: TestContext, assert) {
+    assert.expect(1);
+
+    this.set('onUpdateConfig', (result: { baselineValue: string | number }) => {
+      assert.equal(result.baselineValue, 1, 'onUpdateConfig action is called by baseline input');
+    });
+
+    await render(Template);
+    await fillIn('.goal-gauge-config__baseline-input', '1');
+    await blur('.goal-gauge-config__baseline-input');
+  });
+
+  test('onUpdateConfig goalValue input', async function(this: TestContext, assert) {
+    assert.expect(1);
+
+    this.set('onUpdateConfig', (result: { goalValue: string | number }) => {
+      assert.equal(result.goalValue, 10, 'onUpdateConfig action is called by goal input');
+    });
+
+    await render(Template);
+
+    await run(async () => {
+      await fillIn('.goal-gauge-config__goal-input', '10');
+      await blur('.goal-gauge-config__goal-input');
+    });
```

Comment:
We can probably remove the `run` wrappers for this file
```suggestion
    await fillIn('.goal-gauge-config__goal-input', '10');
    await blur('.goal-gauge-config__goal-input');
```